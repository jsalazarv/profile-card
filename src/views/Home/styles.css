.home-container {
    @apply h-screen;
    @apply flex;
    @apply justify-center;
    @apply items-center;
    @apply p-3;
}