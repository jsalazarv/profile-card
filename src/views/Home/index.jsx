import './styles.css';
import { ProfileCard } from './Components/ProfileCard';
import { ThemeToggle } from '../../components/ThemeToggle';

export const Home = () => {
  return (
    <>
      <ThemeToggle />
      <ProfileCard />
    </>
  );
};
