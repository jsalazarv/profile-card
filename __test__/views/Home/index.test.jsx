import { render } from '@testing-library/react';
import { Home } from '../../../src/views/Home';
import { ProfileCard } from '../../../src/views/Home/Components/ProfileCard/index.jsx';

jest.mock('../../../src/views/Home/Components/ProfileCard');

describe('[Home view]', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('[Snapshots]', () => {
    test('Should content [home-container] class', () => {
      const { container } = render(<Home />);

      expect(container.firstChild).toHaveClass('home-container');
      expect(container).toMatchSnapshot();
    });

    test('Should content [ProfileCard] component', () => {
      render(<Home />);

      expect(ProfileCard).toHaveBeenCalledOnce();
    });
  });
});
