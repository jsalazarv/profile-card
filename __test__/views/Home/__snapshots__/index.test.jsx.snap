// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Home view] [Snapshots] Should content [home-container] class 1`] = `
<div>
  <div
    class="home-container"
  />
</div>
`;
